File: src/main/java/indi/fcwyzzr/minecraft/food_talks/common/mixin/accessors/MobEffectInstanceAccessor.java
package indi.fcwyzzr.minecraft.food_talks.common.mixin.accessors;

import net.minecraft.world.effect.MobEffectInstance;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

@Mixin(MobEffectInstance.class)
public interface MobEffectInstanceAccessor extends Comparable<MobEffectInstance> {
    
    @Accessor(value = "hiddenEffect", remap = false)
    MobEffectInstance getHiddenEffect();
}

File: src/main/kotlin/indi/muxin/food_talks/common/mixin/MobEffectInstanceAccessor.kt
package indi.muxin.food_talks.common.mixin

import net.minecraft.world.effect.MobEffectInstance
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.gen.Accessor

@Mixin(MobEffectInstance::class)
interface MobEffectInstanceAccessor : Comparable<MobEffectInstance> {
    @get:Accessor(value = "hiddenEffect")
    val hiddenEffect: MobEffectInstance?
}